internal/csi-addons/networkfence/fencing.go

```diff
@@ -132,6 +154,90 @@ func (nf *NetworkFence) AddNetworkFence(ctx context.Context) error {
 	return nil
 }
 
+func listActiveClients(ctx context.Context, mdsDaemonID string) ([]ActiveClient, error) {
+	cmd := []string{"tell", fmt.Sprintf("mds.%s", mdsDaemonID), "client", "ls"}
+	stdout, stdErr, err := util.ExecCommand(ctx, "ceph", cmd...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list active clients: %w, stderr: %q", err, stdErr)
+	}
+
+	var activeClients []ActiveClient
+	if err := json.Unmarshal([]byte(stdout), &activeClients); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+	}
+
+	return activeClients, nil
+}
+
+func evictCephFSClient(ctx context.Context, clientIP string, mdsDaemonID string) error {
+	cmd := []string{"tell", fmt.Sprintf("mds.%s", mdsDaemonID), "client", "evict", clientIP}
+	_, stdErr, err := util.ExecCommand(ctx, "ceph", cmd...)
+	if err != nil {
+		return fmt.Errorf("failed to evict client %s: %w, stderr: %q", clientIP, err, stdErr)
+	}
+	log.DebugLog(ctx, "Client %s has been evicted from CephFS\n", clientIP)
+	return nil
+}
+
+func isIPInCIDR(ctx context.Context, ip string, cidr string) bool {
+	// Parse the CIDR block
+	_, ipCidr, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.DebugLog(ctx, "Error parsing CIDR block %s: %v\n", cidr, err)
+		return false
+	}
+
+	// Parse the IP address
+	ipAddress := net.ParseIP(ip)
+	if ipAddress == nil {
+		log.DebugLog(ctx, "Error parsing IP address %s\n", ip)
+		return false
+	}
+
+	// Check if the IP address is within the CIDR block
+	return ipCidr.Contains(ipAddress)
+}
+
+func fetchIP(clientInfo string) (string, error) {
+	parts := strings.Fields(clientInfo)
+	if len(parts) >= 2 {
+		return parts[1], nil
+	}
+	return "", errors.New("unable to extract IP address")
+}
+
+// AddClientEviction blocks access for all the IPs in the CIDR block
+// using client eviction.
+func (nf *NetworkFence) AddClientEviction(ctx context.Context) error {
+	// Fetch active clients
+	activeClients, err := listActiveClients(ctx, "0")
+	if err != nil {
+		return err
+	}
+	// Iterate through CIDR blocks and check if any active client matches
+	for _, cidr := range nf.Cidr {
+		for _, client := range activeClients {
+			clientIP, err := fetchIP(client.Inst)
+			if err != nil {
+				return fmt.Errorf("Error fetching client IP: %s\n", err)
+			}
+			// Check if the clientIP is in the CIDR block
+			if isIPInCIDR(ctx, clientIP, cidr) {
```

Comment:
what if we dont find any activeClients but we still want to blocklist all the IP's in the CIDR?

Reply:
Thanks Madhu, updated. 
For the IPs in CIDR which doesn't have any corresponding active clients, blocklisting like rbd does -> i.e. blocklisting the entire IP by adding to ceph blocklist.
PTAL.